waggle/plugin.py

```diff
@@ -237,43 +239,39 @@ def process_queues_and_events():
         channel.start_consuming()
 
 
-def message_to_amqp(msg: Message) -> Tuple[pika.BasicProperties, bytes]:
+def message_to_amqp(msg: Message) -> bytes:
     # pack metadata into standard amqp message properties
-    properties = pika.BasicProperties(
-        delivery_mode=2,
-        type=msg.name,
-        timestamp=msg.timestamp,
-        reply_to=msg.src)
-
-    # determine content type
-    if isinstance(msg.value, (bytes, bytearray)):
-        body = msg.value
-    elif isinstance(msg.value, Image):
-        # encode as png to send as raw bytes
-        properties.content_type = 'image/png'
-        body = convert_numpy_image_to_png(msg.value.data)
-    else:
-        # attempt to encode all other types as compact json blob
-        properties.content_type = 'application/json'
-        body = json.dumps(msg.value, separators=(',', ':')).encode()
-    
-    return properties, body
+    tmpval=msg.val
+    if msg.enc == "b64":
```

Comment:
We should detect this from the value type so the user doesn't need to explicitly do it. 

Something like:

```python
if isinstance(msg.val, (bytes, bytearray)):
    msg.enc = "b64"
    tmpval = base64.b64encode(msg.val)
```